infra/main.tf

```diff
@@ -121,8 +117,5 @@ resource "google_container_cluster" "my_cluster_config" {
   # Workaround from https://github.com/hashicorp/terraform-provider-google/issues/10782#issuecomment-1024488630
   ip_allocation_policy {
   }
-  node_config {
-    service_account = google_service_account.my_service_account.email
-  }
```

Comment:
_Not_ defining `node_config` like this is what fixed the issue.
Defining `node_config` makes Terraform think, "oh, this is what the node_config should look like... but that's not the case right now so I should deleting this cluster and create a new one" during `terraform destroy`.